### PR TITLE
Adjust for MilliGas and add tests for concat, reverse, format

### PR DESCRIPTION
**Pull request by imalsogreg**

### CODE REVIEW

File: test/Chainweb/Test/Pact/PactMultiChainTest.hs [hunk 1]
```diff
@@ -1044,6 +1045,28 @@ chainweb219UpgradeTest = do
     tryReadMsg = buildBasicGas 1000
         $ mkExec' "(try 1 (read-msg \"somekey\"))"
 
+pact48UpgradeTest :: PactTestM ()
+pact48UpgradeTest = do
+  runToHeight 83 -- TODO: What height do we want pre-fork, post-fork?
```
Comment (jmcardon) [inline test/Chainweb/Test/Pact/PactMultiChainTest.hs:1050]: This height just needs to be a few blocks past the last fork.

Comment (imalsogreg) [reply test/Chainweb/Test/Pact/PactMultiChainTest.hs]: Thanks! Forgot to remove my TODO. fixed now.

**Review by 0xd34df00d — APPROVED** [review]
format/reverse looks good!

File: test/Chainweb/Test/Pact/PactMultiChainTest.hs [hunk 2]
```diff
@@ -1044,6 +1045,28 @@ chainweb219UpgradeTest = do
     tryReadMsg = buildBasicGas 1000
         $ mkExec' "(try 1 (read-msg \"somekey\"))"
 
+pact48UpgradeTest :: PactTestM ()
+pact48UpgradeTest = do
+  runToHeight 83
+
+  -- run block TODO
```
Comment (edmundnoble) [inline test/Chainweb/Test/Pact/PactMultiChainTest.hs:1052]: What's the TODO for?

Comment (imalsogreg) [reply test/Chainweb/Test/Pact/PactMultiChainTest.hs]: Leftover, thanks! Fixed.

**Comment by imalsogreg** [comment]
@edmundnoble In theory this is ready to merge, but, the MacOS CI job is failing with gas discrepancies. Is that a known issue?

**Comment by edmundnoble** [comment]
@imalsogreg no, it isn't.

**Comment by edmundnoble** [comment]
It's not the macOS CI that's failing with gas discrepancies, it's actually all unit tests.

**Comment by edmundnoble** [comment]
@imalsogreg
Just hunted it down, the tests that are failing are the tests for the coin contract. The reason is that the gas involved with coin contract calls changes after this release. What I recommend doing as a mitigation is:
1. updating all of the tests for older coin contract versions to include `DisablePact48` in their execution environment. That is specifically coin and coin-v4.
2. updating the newest coin contract version, which will actually run with the new fork, to make the gas numbers match what the tests report. That's specifically coin-v5.

These tests are located in `pact/coin-contract/coin.repl`, `pact/coin-contract/v4/coin-v4.repl`, and `pact/coin-contract/v5/coin-v5.repl`.

**Pull request merged**